dispatch/dispatch.go

```diff
@@ -182,6 +184,7 @@ func (d *Dispatcher) run(it provider.AlertIterator) {
 				for _, ag := range groups {
 					if ag.empty() {
 						ag.stop()
+						d.marker.DeleteByGroupKey(ag.GroupKey())
```

Comment:
We need to delete the marker when the aggregation group is deleted. There is a similar mechanism in `provider/mem/mem.go`: https://github.com/prometheus/alertmanager/blob/14cbe6301c732658d6fe877ec55ad5b738abcf06/provider/mem/mem.go#L108